### PR TITLE
Update Header

### PR DESCRIPTION
**Pull request by rishab-goswami**
GitHub Issue (If applicable): #129 

## PR Type

What kind of change does this PR introduce?
Enhancement

## What is the current behavior?
The header of the  app currently isn't sticky so it doesn't stay at the top of the page when scrolling
When in the mobile view, the components of the header have some overlap.

## What is the new behavior?
The header bar is sticky, there is a margin between the left side of the header and the right side so there isn't any overlap, the users google profile pic and preferred name are displayed on the header (the name replaces the feedr button and takes the user back to the feed page) and a favorites button has been added (since the favorites system hasn't been implemented, the favorites button changes the view from list to grid as a placeholder)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [ ] Added new tests to the test suite for the feature
- [x] Tested the full test suite, and ensured that the feature does not break current build.
- [x] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [x] Checked licensing of Frameworks/Libraries (if applicable) 

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: client/src/components/Header.jsx [hunk 1]
```diff
@@ -18,59 +21,76 @@ import { Link } from 'react-router-dom';
...
     display: 'flex',
-    marginRight: theme.spacing(3),
-    marginLeft: theme.spacing(3),
+    marginRight: theme.spacing(2),
+    marginLeft: theme.spacing(2),
+    justifyContent: 'space-between',
   },
-  appBarContents: {
+  appBarContentLeft: {
     display: 'flex',
```
Comment (lakshyaoberoi) [inline client/src/components/Header.jsx:29]: Based on the style property name would you need a `justifyContent: 'left'` or a `alignItems='left'` or is it working as intended without these style properties?

Comment (rishab-goswami) [reply client/src/components/Header.jsx:29]: It seems to work fine without them

**Comment by lakshyaoberoi** [comment]
> could we please chuck some padding at the top and bottom of the photo and the search bar
> image

@rishab-goswami can you also move the heart icon to the other side with all the other icons - next to the toggle switch?

**Comment by baja156** [comment]
Also could you please remove the right margin on the toggle theme button, it looks a little weird when i resize the window
image

**Comment by rishab-goswami** [comment]
I made all the changes. Hopefully its good now. I also added some padding to the right of the search bar or else when the window becomes too small, it overlaps with the toggle button

**Pull request merged**